src/PermissionRegistrar.php

```diff
@@ -123,18 +123,19 @@ private function loadPermissions()
                     ->with('roles:id,name,guard_name')
                     ->get();
 
-                if (! method_exists($this->getPermissionClass(), 'getModelFromArray')) {
-                    return $permissions;
-                }
-
                 // make the cache smaller using an array with only required fields
                 return $permissions->map(function ($permission) {
                     return $permission->only('id', 'name', 'guard_name') + ['roles' => $permission->roles->map->only('id', 'name', 'guard_name')->all()];
                 })->all();
             });
             if (is_array($this->permissions)) {
-                $this->permissions = (new Collection($this->permissions))->map(function ($permission) {
-                    return $this->permissionClass::getModelFromArray($permission);
+                $this->permissions = $this->getPermissionClass()::hydrate(
+                    collect($this->permissions)->map(function ($item) {
+                        return collect($item)->only('id', 'name', 'guard_name')->all();
+                    })->all()
+                )
+                ->each(function ($permission, $i) {
+                    $permission->setRelation('roles', $this->getRoleClass()::hydrate($this->permissions[$i]['roles'] ?? []));
```

Comment:
Does this make it as "small" as before? I think it leaves the `['pivot']` sub-array, doesn't it?

Reply:
I don't think that, `hydrate` makes a Collection from array, a couple lines up i make a array from Collection for cache

```php
return $permission->only('id', 'name', 'guard_name') + 
     ['roles' => $permission->roles->map->only('id', 'name', 'guard_name')->all()];
```
 `$permission->roles->map->only('id', 'name', 'guard_name')` returns array without `['pivot']`

Reply:
I guess if `$this->permissions` is never an array, it won't matter.

Reply:
Yes, if `$this->permissions` is a collection, it avoid that code, only with array makes `hydrate` for transform to a colection, It is to support transition, we can remove that condition on next major version, It works same as before(tested on mi applications, no `['pivot']`, minimal expression)

```php
array:358 [▼
  0 => array:4 [▼
    "id" => 197
    "name" => "view_audit"
    "guard_name" => "web"
    "roles" => array:3 [▼
      0 => array:3 [▼
        "id" => 0
        "name" => "System"
        "guard_name" => "web"
      ]
      1 => array:3 [▶]
      2 => array:3 [▶]
    ]
  ]
```